File: skills/tooling-java-generator/examples/CliExample.java
package com.example.cli;

import picocli.CommandLine;
import picocli.CommandLine.*;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.concurrent.Callable;

/**
 * Java CLI Example: file line counter with picocli
 * Demonstrates: CLI argument parsing, file I/O, exit codes
 */
@Command(name = "linecount", mixinStandardHelpOptions = true, version = "1.0")
public class CliExample implements Callable<Integer> {

    @Parameters(index = "0", description = "File to count")
    private Path file;

    @Override
    public Integer call() throws IOException {
        long lines = Files.lines(file).count();
        System.out.printf("%d lines in %s%n", lines, file.getFileName());
        return 0;
    }

    public static void main(String[] args) {
        System.exit(new CommandLine(new CliExample()).execute(args));
    }
}

File: skills/tooling-csharp-generator/examples/CliExample.cs
using System.CommandLine;

namespace Example.Cli;

/// <summary>
/// C# CLI Example: file line counter
/// Demonstrates: System.CommandLine, file I/O, async/await, exit codes
/// </summary>
public class Program
{
    public static async Task<int> Main(string[] args)
    {
        var fileArg = new Argument<FileInfo>("file", "File to count lines");
        var rootCommand = new RootCommand("Count lines in a file") { fileArg };

        rootCommand.SetHandler(async (FileInfo file) =>
        {
            var lines = await File.ReadAllLinesAsync(file.FullName);
            Console.WriteLine($"{lines.Length} lines in {file.Name}");
        }, fileArg);

        return await rootCommand.InvokeAsync(args);
    }
}
